LoopKit/DosingDecisionStore.swift

```diff
@@ -227,6 +230,33 @@ public struct StoredDosingDecision {
             self.date = date
         }
     }
+
+    public struct DeviceSettings: Codable, Equatable {
+        let name: String
+        let systemName: String
+        let systemVersion: String
+        let model: String
+        let modelIdentifier: String
+        let batteryLevel: Float?
+        let batteryState: BatteryState?
+
+        public init(name: String, systemName: String, systemVersion: String, model: String, modelIdentifier: String, batteryLevel: Float? = nil, batteryState: BatteryState? = nil) {
+            self.name = name
+            self.systemName = systemName
+            self.systemVersion = systemVersion
+            self.model = model
+            self.modelIdentifier = modelIdentifier
+            self.batteryLevel = batteryLevel
+            self.batteryState = batteryState
+        }
+
+        public enum BatteryState: String, Codable {
```

Comment:
[nit] Did you opt  for this for the easy `Codable` implementation, right?  Otherwise, `BatteryState` could be:
```swift
        public enum BatteryState: String, Codable {
            case unknown
            case unplugged(level: Double)
            case charging(level: Double)
            case full
}
```
But again, this is a [nit].

Reply:
UIDevice reports battery state and battery level independently. Without further documentation to the contrary, I figured that when battery state was unknown or full it was still possible to report a valid, non-1.0 battery level.